device/nrf51822.go

```diff
@@ -0,0 +1,91 @@
+package device
+
+import "log"
+
+type operation byte
+
+const (
+	START_DFU                   operation = 0x01 //Can I use iota here?
```

Comment:
no need to repeat the type. This is equivalent to:

```go
const (
        START_DFU operation = 0x01
        INITIALISE_DFU = 0x02
        ...
)
```

Also why are you using hex here? Is this defined by the device type itself I guess? You can use iota like:

```go
foo bar = iota + 1
```

But you later skip out digits, e.g. 9, 10, 11, 12, 13, 14, 15 so it might not be appropriate (is this intentional is this meant to be 0x10, 0x11 in hex?)